proxy/proxy.go

```diff
@@ -37,31 +37,20 @@ func init() {
 	}
 }
 
-type zstdCloser struct {
-	*zstd.Decoder
-}
-
-func (zc *zstdCloser) Close() error {
-	zc.Decoder.Close()
-	return nil
-}
-
 func Decompress(rdr io.ReadCloser, encoding string) (io.ReadCloser, error) {
 	switch encoding {
 	case "gzip":
 		decomp, err := gzip.NewReader(rdr)
 		if err != nil {
 			return nil, err
 		}
-		defer decomp.Close()
```

Comment:
That should make sense, right? I'm afraid this might even be a bug in current main.

Reply:
It does